Cargo.toml

```diff
@@ -46,10 +46,15 @@ std = [
 [dependencies]
 blake3 = { version = "1.5", default-features = false }
 clap = { version = "4.5", features = ["derive"], optional = true }
+getrandom = { version = "0.2", features = ["js"] }
+num = { version = "0.4.1", default-features = false, features = ["alloc", "rand"] }
+num-complex = { version = "0.4.4", default-features = false }
+rand = { version = "0.8.5", default-features = false, features = ["getrandom"] }
```

Comment:
1. Changed
2. Removed
3. It seems that `OsRng` needs it.
4. Done

Reply:
@bobbinth One thing to note is that with the `getrandom` feature enabled (and I doubt there is any way to escape needing it without rolling our own RNG), I'm pretty sure Wasm builds will need to use the `wasm32-wasi` target rather than `wasm32-unknown-unknown`. That isn't a big deal, but just something to be aware of, since without WASI, the underlying host functionality expected by `getrandom` will be missing. Probably will just require tweaking how Wasm builds are shipped to include the browser shims needed for it.

Reply:
Indeed! See [here](https://github.com/rust-random/getrandom/blob/5f0701faba5b83ebf144af9973582904f60849b7/src/lib.rs#L58)

Reply:
> For std builds, we shouldn't need getrandom (I'm assuming when std feature is enabled for rand getrandom implied internally).

IIRC that's true, yes.

> For no_std builds, the only thing we get from rand is the Rng trait. The user is then responsible for suppling the implementation of the Rng trait (whether from getrandom or from somewhere else) when using the relevant functions. So, in this case, we don't need getrandom dependency either.

That's a totally viable approach, though I would take this a step further:

* Add a feature to make use of `getrandom` (as an optional dependency), as well as a feature to enable the `getrandom/js` feature so that it can be used when compiling to Wasm. Add `dep:getrandom` as an implied feature when the `std` feature is enabled.
* Have the base API layer require passing in an implementation of `Rng` as configuration.
* Add a set of wrapper APIs that are enabled when the `getrandom` feature is present, that pass in the `getrandom` RNG implementation to the base APIs.

This approach could also be extended to allow enabling different features for different commonly-used RNGs, e.g. the default RNG supplied/chosen by the wrapper functions depends on which feature is enabled, e.g. `getrandom` or something else.

Without those features, the base API is still usable, but simply requires providing an RNG everywhere.

That said, I'm not sure it's actually necessary to do this versus always using `getrandom` and just making the Wasm build work, but I can see value in punting that to downstream crates.

Reply:
> I would take this a step further:
> 
> * Add a feature to make use of `getrandom` (as an optional dependency), as well as a feature to enable the `getrandom/js` feature so that it can be used when compiling to Wasm. Add `dep:getrandom` as an implied feature when the `std` feature is enabled.
> * Have the base API layer require passing in an implementation of `Rng` as configuration.
> * Add a set of wrapper APIs that are enabled when the `getrandom` feature is present, that pass in the `getrandom` RNG implementation to the base APIs.

Agreed - but I would leave this for a future PR (this one has been open for quite some time already). For now, I just removed `getrandom` dependency.